### PR TITLE
Initialize local value table in path dependent branch folding for IntConstOpnds

### PR DESCRIPTION
**Pull request by meg-gupta**
We were not adding value of IntConstOpnd in the local value table, nor were we initializing it to null.
This caused to incorrectly read value from the global value table, which might be inaccurate.

### CODE REVIEW

**Comment by meg-gupta** [comment]
The benchmark turns out to be too big to reduce.

**Comment by meg-gupta** [comment]
The basic pattern that was faulty is : 

L1 : 
 badVar = 1;

..

File: lib/Backend/FlowGraph.cpp [hunk 1]
```diff
@@ -4646,7 +4646,18 @@ BasicBlock::CheckLegalityAndFoldPathDepBranches(GlobOpt* globOpt)
             StackSym* src1Sym = instr->GetSrc1()->GetStackSym();
             FindValueInLocalThenGlobalValueTableAndUpdate(globOpt, localSymToValueMap, instr, instr->GetDst()->GetSym(), src1Sym);
         }
-        else if (!instr->GetSrc1()->IsIntConstOpnd())
+        else if (instr->GetSrc1()->IsIntConstOpnd())
+        {
+            Value **localValue = localSymToValueMap->FindOrInsertNew(instr->GetDst()->GetSym());
+            *localValue = globOpt->NewValue(IntConstantValueInfo::New(globOpt->alloc, instr->GetSrc1()->AsIntConstOpnd()->AsInt32()));
```
Comment (rajatd) [inline lib/Backend/FlowGraph.cpp]: Instead of creating a new value, can we use the existing value for the int constant? (GlobOpt::GetIntConstantValue)

Comment (meg-gupta) [reply lib/Backend/FlowGraph.cpp]: Yes, thank you.

**Comment by meg-gupta** [comment]
@dotnet-bot test OSX static_osx_osx_test

**Pull request merged**